parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java

```diff
@@ -299,19 +292,42 @@ public boolean isDirty() {
         return pm.isDirty();
     }
 
-    public void addMetricsTo(PCMetrics.PCMetricsBuilder<?, ? extends PCMetrics.PCMetricsBuilder<?, ?>> metrics) {
-        metrics.partitionMetrics(getPm().getMetrics());
-        metrics.shardMetrics(getSm().getMetrics());
-        metrics.workManagerMetrics(getMetrics());
+    private void initMetrics() {
+        waitingRecordsNumberGauge = PCMetrics.getInstance().gaugeFromMetricDef(PCMetricsDef.WAITING_RECORDS,
+                this, WorkManager::getNumberOfWorkQueuedInShardsAwaitingSelection);
+        inflightRecordsNumberGauge = PCMetrics.getInstance().gaugeFromMetricDef(PCMetricsDef.INFLIGHT_RECORDS,
+                this, WorkManager::getNumberRecordsOutForProcessing);
     }
 
-    public void enrichWithIncompletes(Map<TopicPartition, PCMetrics.PCPartitionMetrics> metrics) {
-        getPm().enrichWithIncompletes(metrics);
+    private void initCounterMetrics(Collection<TopicPartition> partitions) {
+        partitions.forEach(topicPartition -> {
+            if (!succeededRecordsCounters.containsKey(topicPartition)) {
+                succeededRecordsCounters.put(topicPartition, PCMetrics.getInstance().getCounterFromMetricDef(PCMetricsDef.PROCESSED_RECORDS, getWorkManagerCounterTags(topicPartition)));
+            }
+            if (!failedRecordsCounters.containsKey(topicPartition)) {
+                failedRecordsCounters.put(topicPartition, PCMetrics.getInstance().getCounterFromMetricDef(PCMetricsDef.FAILED_RECORDS, getWorkManagerCounterTags(topicPartition)));
+            }
+        });
+    }
+
+    private void safelyIncrementCounter(Map<TopicPartition, Counter> counterMap, TopicPartition topicPartition) {
```

Comment:
Not crazy about this name. Just a nitpick

```suggestion
    private void incrementCounterIfPresent(Map<TopicPartition, Counter> counterMap, TopicPartition topicPartition) {
```